### ISSUE TITLE
It takes sbt a long time to consume long commands issued by IntelliJ

### ISSUE THREAD

**Issue by RafalSumislawski** [sbt/sbt#6644]
## steps

- clone https://github.com/http4s/http4s (I'm using it as an example because it's large and it's open source, but other projects are affected the same way)
- Import it in IntelliJ
- Enable using sbt shell for builds and imports in IntelliJ.
- open sbt shell panel in IntelliJ
- Ask IntelliJ to build the project
- Watch a "slide show" as sbt slowly (~10 seconds) consumes the command issued by IntelliJ. (The command has a few thousands of characters)
- Also notice 100% utlisation of one CPU core

sbt version 1.5.5 (1.4.0-1.5.5 are affected)
OS: MacOS 11.5.2
IntelliJ 2021.2.1 CE (but the issue isn't related to specific IntelliJ version)

## problem

It appears that starting with 1.4.0 sbt started taking much more time to read characters from sbt console. 
The issue did not appear with 1.3.x, and it seems unrelated to the version of IntelliJ in use.

On closer inspection with a profiler it appear that the `sbt.internal.util.Terminal.ConsoleTerminal#isEchoEnabled` is being called for each character read and that on each call it spawns and waits for a new process.

<img width="1056" alt="image" src="https://user-images.githubusercontent.com/9465998/131385839-1426155e-ec7b-49e2-8d85-f7790312b046.png">

## expectation

sbt should consume the input in insignificant amount of time like it did in 1.3.x

## notes

Handling of `echo` has been reworked in jline3, so maybe the issue wouldn't appear if jline3 was used in sbt.

**Comment by eed3si9n** [comment]
Could you try using sbtn (https://github.com/sbt/sbtn-dist/releases)? We built a native thin client that can talk to a running instance of sbt. This is also available as `sbt --client compile`

sbt's use of JLine is only for our own shell and we don't really test any other use cases, so I don't think it's our responsibility to keep functional parity across the versions.

**Comment by RafalSumislawski** [comment]
Hi @eed3si9n 

As long as I want to use IntelliJ, using sbtn is not a solution for me. 

Of course, I understand that it's not `sbt`'s responsibility to test and keep compatibility with IntelliJ. I'm trying to use the two tools together and so it's in my interest to find where this can be fixed. I decided to report the issue here first, as opposed to reporting it first to JetBrains, because the strange thing (spawning thousands of processes) seems to be happening in sbt, not in IntelliJ's code. 

Today I had some more time to ask myself the question "Why this issue affects IntelliJ's shell, but doesn't affect regular sbt shell". And that's how I learned about the existence of `shell` and `oldshell` in sbt. It turns out IntelliJ's shell behaviour is basically the oldshell behaviour. In other words I can now reproduce the issue with pure sbt (no IntelliJ involved) with the following steps:

* run `sbt oldshell` in any project
* paste a long command (anything) into the shell.
I use this as the test "command":
```
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
```
* watch the "animation" of the input appearing in the shell
